src/overlays/actors/ovl_Dm_Stk/z_dm_stk.c

```diff
@@ -808,75 +1009,79 @@ void DmStk_Init(Actor* thisx, GlobalContext* globalCtx) {
+        this->objectStk3ObjectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_STK3);
+        if (this->objectStkObjectIndex < 0) {
             Actor_MarkForDeath(&this->actor);
         }
 
-        this->unk_328 = 0;
-        this->unk_339 = 0;
-        this->unk_32C = 1;
-        this->unk_2E0 = 3;
-        this->unk_2E8 = globalCtx->lightCtx.unk7;
-        this->unk_2EC = globalCtx->lightCtx.unk8;
-        this->unk_2F0 = globalCtx->lightCtx.unk9;
+        this->tatlMessageTimer = 0;
+        this->hitCount = 0;
+        this->maskType = SK_MASK_TYPE_NORMAL;
+        this->animationId = SK_ANIMATION_IDLE;
+        this->fogR = globalCtx->lightCtx.unk7;
+        this->fogG = globalCtx->lightCtx.unk8;
+        this->fogB = globalCtx->lightCtx.unk9;
 
         if ((globalCtx->sceneNum == SCENE_LOST_WOODS) && (gSaveContext.sceneSetupIndex == 1)) {
-            this->unk_2E4 = 0;
-            this->unk_2F8 = 0;
-            this->unk_2FC = 1000;
-            this->unk_300 = 1.0f;
-            this->actionFunc = func_80AA1704;
+            this->alpha = 0;
+            this->fogN = 0;
+            this->fogF = 1000;
+            this->fogScale = 1.0f;
+            this->actionFunc = DmStk_DoNothing;
         } else if (globalCtx->sceneNum == SCENE_OKUJOU) {
-            this->unk_2E4 = 255;
-            this->unk_2F8 = 996;
-            this->unk_2FC = 1000;
-            this->unk_300 = 0.7f;
-            this->unk_335 = 0;
+            this->alpha = 255;
+            this->fogN = 996;
+            this->fogF = 1000;
+            this->fogScale = 0.7f;
+            this->hasBeenHit = false;
 
             Collider_InitCylinder(globalCtx, &this->collider);
 
             if (gSaveContext.save.entranceIndex == 0x2C00) {
                 if (gSaveContext.sceneSetupIndex == 0) {
                     if (gSaveContext.unk_3DD0[3] == 0) {
+                        // Starts a 5 minute (300 second) timer until the moon falls.
                         func_8010E9F0(3, 300);
                         XREG(80) = 200;
                         XREG(81) = 115;
                     }
 
                     if (gSaveContext.save.inventory.items[SLOT_OCARINA] == ITEM_NONE) {
                         sCylinderInit.base.colType = COLTYPE_WOOD;
-                        this->actionFunc = func_80AA18D8;
+                        this->actionFunc = DmStk_ClockTower_StartFirstTimeIntroCutscene;
                     } else {
                         sCylinderInit.base.colType = COLTYPE_WOOD;
-                        this->actionFunc = func_80AA1998;
+                        this->actionFunc = DmStk_ClockTower_StartSubsequentTimesIntroCutscene;
                     }
 
                 } else if (gSaveContext.sceneSetupIndex == 3) {
-                    this->unk_2E0 = 38;
+                    this->animationId = SK_ANIMATION_FLOATING_ARMS_CROSSED;
                     if (gSaveContext.unk_3DD0[3] == 0) {
```

Comment:
I wonder if it's possible to name this SaveContext variable, then.

Reply:
Unfortunately, it's used all over the place. For example, EnMtTag sets `unk_3DD0[4]` to 6 when anyone crosses the finish line in the Goron Race. Skull Kid is the only actor so far to use `unk_3DD0[3]`, though.

Reply:
Hmm, annoying, will probably turn out to just be general-purpose stuff, then.